code/glTF2Asset.h

```diff
@@ -244,15 +242,17 @@ namespace glTF2
     };
 
     //! Values for the Sampler::magFilter field
-    enum SamplerMagFilter
+    enum class SamplerMagFilter: unsigned int
```

Comment:
I "had" to use strongly typed enums as I needed a way to set a unset state; simply checking an uninitialized enum value was undefined behaviour. Is this okay? Will assimp maintainers be okay with this form? Or is it overkill?

Reply:
I personally prefer `enum class` for various reasons, but I think you could still do this with a regular `enum`, no? It's very common to define something like the following:
```c++
enum Something {
    Something_UNKOWN = 0,
    Something_ELSE = 1
}
```
Also, as a stylistic matter, one of the nice things about `enum class`es is that their members are scoped, so you could do something like:
```c++
    enum class SamplerMagFilter: unsigned int
    {
        UNSET = 0,
        Nearest = 9728,
        Linear = 9729
    };
```
That is, omitting the `SamplerMagFilter_` prefix, which is only used in the regular `enum` because its members aren't scoped, which causes problems in situations like the following:
```c++
// my_code.h
enum MyBool {
    TRUE,
    FALSE
}

// some_library.h
enum LibraryBool {
    TRUE,
    FALSE
}

// main.cpp
#include "my_code.h"
#include "some_library.h"
// -> compiler complains about re-declaration of `TRUE` and `FALSE`
```

Reply:
Ah, right. I think that's why I ultimately chose "enum class" because they had the side-effect of being scoped, so I could re-use `UNSET` across different enums